### PR TITLE
49 crp standardize control class

### PR DESCRIPTION
**Pull request by mesmatyi**
Added control class cleanup

### CODE REVIEW

**Review by AnonymDavid — CHANGES_REQUESTED** [review]
Some coding rules changes:
- the class definition should be in a separate .hpp header file
- also all the includes (except the header) should be in the header file
- some naming conventions
- also file names should use cammelCase too

**Comment by mesmatyi** [comment]
@AnonymDavid, done

**Review by AnonymDavid — CHANGES_REQUESTED** [review]
Lat controller still does not have its headers and includes separated to a header file.

**Comment by mesmatyi** [comment]
i guess it's good for now @AnonymDavid

File: crp_APL/controllers/ctrl_vehicle_control_lat_compensatory/include/ctrl_vehicle_control_lat_compensatory/ctrl_vehicle_control_lat_compensatory.hpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+                void trajCallback(const autoware_planning_msgs::msg::Trajectory input_msg);
+                void egoVehicleCallback(const crp_msgs::msg::Ego input_msg);
+
+                void loop();
+
+                rclcpp::TimerBase::SharedPtr timer_;
+                rclcpp::Publisher<autoware_control_msgs::msg::Lateral>::SharedPtr m_pub_cmd;
+
+                rclcpp::Subscription<autoware_planning_msgs::msg::Trajectory>::SharedPtr traj_sub;
```
Comment (gfigneczi1) [inline crp_APL/controllers/ctrl_vehicle_control_lat_compensatory/include/ctrl_vehicle_control_lat_compensatory/ctrl_vehicle_control_lat_compensatory.hpp:49]:
```suggestion
                rclcpp::Subscription<autoware_planning_msgs::msg::Trajectory>::SharedPtr m_traj_sub_;
```

File: crp_APL/controllers/ctrl_vehicle_control_lat_compensatory/include/ctrl_vehicle_control_lat_compensatory/ctrl_vehicle_control_lat_compensatory.hpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                rclcpp::Subscription<crp_msgs::msg::Ego>::SharedPtr ego_vehicle_sub;
```
Comment (gfigneczi1) [inline crp_APL/controllers/ctrl_vehicle_control_lat_compensatory/include/ctrl_vehicle_control_lat_compensatory/ctrl_vehicle_control_lat_compensatory.hpp:50]:
```suggestion
                rclcpp::Subscription<crp_msgs::msg::Ego>::SharedPtr m_egoVehicle_sub_;
```

File: crp_APL/controllers/ctrl_vehicle_control_lat_compensatory/include/ctrl_vehicle_control_lat_compensatory/ctrl_vehicle_control_lat_compensatory.hpp [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                autoware_control_msgs::msg::Lateral ctrlCmd;
```
Comment (gfigneczi1) [inline crp_APL/controllers/ctrl_vehicle_control_lat_compensatory/include/ctrl_vehicle_control_lat_compensatory/ctrl_vehicle_control_lat_compensatory.hpp:51]:
```suggestion
                autoware_control_msgs::msg::Lateral m_ctrlCmdMsg;
```

File: crp_APL/controllers/ctrl_vehicle_control_lat_compensatory/src/ctrl_vehicle_control_lat_compensatory.cpp [hunk 4]
```diff
@@ -1,157 +1,123 @@
...
 
-class CtrlVehicleControl : public rclcpp::Node
+
+crp::apl::CtrlVehicleControlLat::CtrlVehicleControlLat() : Node("CtrlVehicleControlLat")
+{
+    timer_ = this->create_wall_timer(std::chrono::milliseconds(33), std::bind(&CtrlVehicleControlLat::loop, this));  
+    m_pub_cmd = this->create_publisher<autoware_control_msgs::msg::Lateral>("/control/command/control_cmdLat", 30);
+
+    traj_sub = this->create_subscription<autoware_planning_msgs::msg::Trajectory>("/plan/trajectory", 10, std::bind(&CtrlVehicleControlLat::trajCallback, this, std::placeholders::_1));
+    ego_vehicle_sub = this->create_subscription<crp_msgs::msg::Ego>("/ego", 10, std::bind(&CtrlVehicleControlLat::egoVehicleCallback, this, std::placeholders::_1));
```
Comment (gfigneczi1) [inline crp_APL/controllers/ctrl_vehicle_control_lat_compensatory/src/ctrl_vehicle_control_lat_compensatory.cpp:13]: accordingly per the other comments

File: crp_APL/ctrl_vehicle_control/include/ctrl_vehicle_control/ctrl_vehicle_control.hpp [hunk 5]
```diff
@@ -13,31 +13,31 @@
...
+                rclcpp::Subscription<autoware_control_msgs::msg::Lateral>::SharedPtr m_sub_controlLat_;
+                rclcpp::Subscription<autoware_control_msgs::msg::Longitudinal>::SharedPtr m_sub_controlLong_;
+                
+                rclcpp::Publisher<autoware_control_msgs::msg::Control>::SharedPtr m_pub_control_;
+                rclcpp::Publisher<geometry_msgs::msg::Twist>::SharedPtr m_pub_twist_;
+
+                rclcpp::TimerBase::SharedPtr m_timer_;    
+
+                geometry_msgs::msg::Twist m_twistMsg;
+                autoware_control_msgs::msg::Control ctrl_msg;
```
Comment (gfigneczi1) [inline crp_APL/ctrl_vehicle_control/include/ctrl_vehicle_control/ctrl_vehicle_control.hpp:38]:
```suggestion
                autoware_control_msgs::msg::Control m_ctrlCmdMsg;
```

File: crp_APL/ctrl_vehicle_control/src/ctrl_vehicle_control.cpp [hunk 6]
```diff
@@ -10,28 +10,59 @@ crp::apl::ControlHandler::ControlHandler() : Node("ControlHandler")
...
+
+
+    m_twistMsg.linear.x = msg->velocity;
+    ctrl_msg.longitudinal.velocity = msg->velocity;
 }
 
 void crp::apl::ControlHandler::run()
 {
-    m_pub_twist_->publish(twist_msg);
+    ctrl_msg.stamp = this->now();
```
Comment (gfigneczi1) [inline crp_APL/ctrl_vehicle_control/src/ctrl_vehicle_control.cpp:63]: shall we add the same stamp to the twist message as well?

**Review by gfigneczi1 — CHANGES_REQUESTED** [review]
minor coding rule related changes needed

File: crp_APL/ctrl_vehicle_control/src/ctrl_vehicle_control.cpp (hunk 6 above)
Comment (mesmatyi) [reply crp_APL/ctrl_vehicle_control/src/ctrl_vehicle_control.cpp:63]: the twist msg has no time stamp

**Comment by mesmatyi** [comment]
@gfigneczi1 check again, corrected the required changes

File: crp_APL/controllers/ctrl_vehicle_control_lat_compensatory/include/ctrl_vehicle_control_lat_compensatory/ctrl_vehicle_control_lat_compensatory.hpp [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        class CtrlVehicleControlLat : public rclcpp::Node
+        {
+            public:
+                CtrlVehicleControlLat();
+
+            private:
+                // VARIABLES
+                crp::apl::ControlInput input;
+                crp::apl::ControlOutput output;
+                crp::apl::ControlParams params;
```
Comment (gfigneczi1) [inline crp_APL/controllers/ctrl_vehicle_control_lat_compensatory/include/ctrl_vehicle_control_lat_compensatory/ctrl_vehicle_control_lat_compensatory.hpp:37]: please ezeket is írd már át m_ -el kezdődőre... bocsánat, csak hogy legyen egységes

**Review by gfigneczi1 — APPROVED** [review]
azt az egy kommentet leszámítva fasza

**Pull request merged**